src/lf_mask.rs

```diff
@@ -128,15 +129,37 @@ fn decomp_tx(
         let lw = cmp::min(2, t_dim.lw);
         let lh = cmp::min(2, t_dim.lh);
 
+        let txa = txa.as_mut_ptr() as *mut [[[[u8; 32]; 32]; 2]; 2];
         CaseSet::<16, false>::one((), t_dim.w as usize, x0, |case, ()| {
             for y in 0..t_dim.h as usize {
-                case.set(&mut txa[0][0][y0 + y], lw);
-                case.set(&mut txa[1][0][y0 + y], lh);
```

Comment:
I think we can do this a lot simpler actually.  Here `txa[i][j][j]` is a `[u8; 32]`, so we know the length.  We can just set it to `[lw; 32]` for example, no casting needed, and no `case.set` needed.  So if we have the type be `txa: &mut Align16<[[[MaybeUninit<[u8; 32]>; 32]; 2]; 2]>`, then we can do `txa[0][0][y0 + y].write([lw; 32])`.  For the final `.assume_init()`, it'll probably be easier to just `mem::transmute` it if we have the `MaybeUninit` a few dimensions inside.

Reply:
Don't we have an offset and len involved? We can't just write 32 bytes every time.

Reply:
Oh yeah, I misunderstood.